src/internal/grpc/headers-interceptor.ts

```diff
@@ -15,7 +15,7 @@ export class Header {
   }
 }
 
-export class HeaderInterceptor {
+export class HeaderInterceptorProvider {
```

Comment:
For my benefit, what was the thought behind the rename?

Reply:
because this class doesn't actually implement Interceptor, it just provides a function that will create one and return it.